platform-modules/batch-models/src/main/scala/org/ekstep/analytics/job/report/StateAdminReportJob.scala

```diff
@@ -80,14 +82,38 @@ object StateAdminReportJob extends optional.Application with IJob with BaseRepor
         // Purge the directories after copying to the upload staging area
         fSFileUtils.purgeDirectory(detailDir)
         fSFileUtils.purgeDirectory(summaryDir)
+        // Only claimed used
+        val claimedShadowDataSummaryDF = claimedShadowUserDF.groupBy("channel")
+          .pivot("claimstatus").agg(count("claimstatus")).na.fill(0)
+
+        saveUserValidatedSummaryReport(claimedShadowDataSummaryDF, s"$summaryDir")
+        saveUserDetailsReport(claimedShadowUserDF.toDF(), s"$detailDir")
+
+        fSFileUtils.renameReport(detailDir, renamedDir, ".csv", "validated-user-detail")
+        fSFileUtils.renameReport(summaryDir, renamedDir, ".json", "validated-user-summary")
+
+        // Purge the directories after copying to the upload staging area
+        fSFileUtils.purgeDirectory(detailDir)
+        fSFileUtils.purgeDirectory(summaryDir)
+
+        val blockDataWithSlug = stateAdminReport.generateGeoBlockData()
+        val resultDF = blockDataWithSlug.join(claimedShadowUserDF, blockDataWithSlug.col("externalid") === (claimedShadowUserDF.col("orgextid")),"left")
+        resultDF.groupBy(col("slug"),col("District name").as("districtName")).
+          agg(countDistinct("Block id").as("blocks"),countDistinct(claimedShadowUserDF.col("orgextid")).as("schools"), count("userextid").as("registered")).write
+          .partitionBy("slug")
+          .mode("overwrite")
+          .json(s"$summaryDir")
+
+        fSFileUtils.renameReport(summaryDir, renamedDir, ".json", "validated-user-summary-district")
+        fSFileUtils.purgeDirectory(summaryDir)
         shadowUserDF.distinct()
     }
 
     private def getChannelSlugMap()(implicit sparkSession: SparkSession): Map[String, String] = {
-        val channelSlugMap: Map[String, String] = loadData(sparkSession, Map("table" -> "organisation", "keyspace" -> sunbirdKeyspace), None)
-          .select(col("channel"), col("slug")).where(col("isrootorg") && col("status").===(1))
-          .collect().groupBy(f => f.get(0).asInstanceOf[String]).mapValues(f => f.head.get(1).asInstanceOf[String]);
-        return channelSlugMap
+      val channelSlugMap: Map[String, String] = loadData(sparkSession, Map("table" -> "organisation", "keyspace" -> sunbirdKeyspace), None)
+        .select(col("channel"), col("slug")).where(col("isrootorg") && col("status").===(1))
+        .collect().groupBy(f => f.get(0).asInstanceOf[String]).mapValues(f => f.head.get(1).asInstanceOf[String]);
+      return channelSlugMap
```

Comment:
Org data frame is loaded twice within the job. It should not happen.

Reply:
Taken care, now it will be called only once.